test/integration/kubernetes_deploy_test.rb

```diff
@@ -1053,4 +1053,31 @@ def test_raise_on_yaml_missing_kind
       "  datapoint: value1"
     ], in_order: true)
   end
+
+  def test_hpa_can_be_successful
+    skip if KUBE_SERVER_VERSION < Gem::Version.new('1.8.0')
+    assert_deploy_success(deploy_fixtures("hpa"))
+    assert_logs_match_all([
+      "Deploying resources:",
+      "HorizontalPodAutoscaler/hello-hpa (timeout: 180s)",
+      %r{HorizontalPodAutoscaler/hello-hpa\s+Configured}
+    ])
+  end
+
+  def test_hpa_times_out_when_no_matching_deployment
+    skip if KUBE_SERVER_VERSION < Gem::Version.new('1.8.0')
+    assert_deploy_failure(deploy_fixtures("hpa", subset: ["hpa.yml"]), :timed_out)
+    assert_logs_match_all([
+      "Deploying HorizontalPodAutoscaler/hello-hpa (timeout: 180s)",
```

Comment:
This test is guaranteed to take at least 180s 😢 
There's some value in asserting that the message displayed on timeout is helpful, but I'm not sure it's worth 3 minutes of CI. Timeouts are fallback behaviour, not something we're generally striving for. I'd be inclined to replace this with a test for a case that actually fails.

Reply:
I'm having a hard time figuring out how to make the HPA pass validation, have AbleToScale be true, and have ScalingActive false with out the cause being in `RECOVERABLE_CONDITIONS`. 

The k8s tests (https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/podautoscaler/horizontal_test.go) weren't very instructional, any suggestions?

Reply:
I can't find anything useful either. Kinda weird that so many conditions have been dedicated to situations that are not possible to create from the outside. My final observation is that in playing around with this locally, there seems to be a substantial delay before the conditions get populated, so maybe it would be safe to fail on `ScalingActive/FailedGetThingMetric` after all, i.e. the benefit of failing fast for those misconfigurations would outweigh the (small) possibility of a race condition causing spurious failures. Without much direct HPA experience, I'm not really sure what is best. 

If we can't reproduce a failure scenario, maybe we can set a short timeout on the hpa resource? That won't really work if the condition message we'd look for in the logs can also take 3 minutes to appear though.

Reply:
How would you feel about using `Timecop.scale` to make this test faster in wallclock time? 

In general I dislike trading correctness for an optimization, failing fast.